doc/modules/classes.rst

```diff
@@ -143,43 +143,58 @@ Classes
    covariance.oas
    covariance.graph_lasso
 
+.. _model_selection_ref:
```

Comment:
I'm -1 for keeping the old modules listed here, but don't object to having their docs generated (by putting automodule somewhere hidden, say "deprecated.rst") with a clear deprecation-oriented docstring, if that's what you're more-or-less going for, @amueller. I'm not sure whether there's an easy way to set `rel="canonical"` to point to new equivalents, if that's your concern @vene.